### ISSUE TITLE
Support Excluding through expression

### ISSUE THREAD

**Issue by acouch** [rhysd/actionlint#414]
Running into an issue where I'd like to excldue a matrix entry via an expression, but it flags that the return value isn't one of the matrix combos:

```
value "${{ (github.event_name == 'release' && 'prod') || 'dev' }}" in "exclude" does not match in matrix "envs" combinations. possible values are "dev", "staging", "prod"
```

![image](https://github.com/rhysd/actionlint/assets/512243/c03f6d42-1020-4a56-a86a-4c11bc2d3b46)

Here is an example file:

```yml
name: Test Matrix

run-name: Deploy ${{ github.ref_name }} 

on: [push]

jobs:
  deploy-infra:
    name: Deploy Infrastructure
    runs-on: ubuntu-latest
    strategy:
      matrix:
        directory: ["database", "service"]
        envs: ['dev', 'staging', 'prod']
        exclude:
          - envs: ${{ (github.event_name == 'release' && 'prod') || 'dev' }}
    steps:
      - name: Show matrix
        run: |
          echo ${{ matrix.directory }} ${{ matrix.envs }}
```

Any assistance or pointers appreciated.

**Comment by rhysd** [comment]
Thank you for reporting this. This is a bug that actionlint does not recognize expressions in `strategy.matrix.exclude`.
